sky/resources.py

```diff
@@ -834,13 +834,15 @@ def _try_validate_image_id(self) -> None:
             # Check the image exists and get the image size.
             # It will raise ValueError if the image does not exist.
             image_size = self.cloud.get_image_size(image_id, region)
-            if image_size > self.disk_size:
+            if image_size >= self.disk_size:
```

Comment:
`=` seems unexpected. Do you mean you cannot create a machine with the same size as the image? I think we did try creating image with 256GB and launch a VM from the image with the same disk size?

Reply:
Users should be able to create instances with the same size as the image in the latest commit. I’ve updated the PR description to include additional tests for it.